backend/src/tools/process_md.py

```diff
@@ -0,0 +1,47 @@
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.docstore.document import Document as LangchainDocument
+from langchain_community.document_loaders import DirectoryLoader
+
+from src.tools.recursive_chunk import recursive_chunking
+
+md_seperators = [
+    "\n#{1,6} ",
+    "```\n",
+    "\n\\*\\*\\*+\n",
+    "\n---+\n",
+    "\n___+\n",
+    "\n\n",
+    "\n",
+    " ",
+    "",
+    "```"
+]
+
+def chunk_markdown(embeddings_model_name: str , files_path: str, chunk_size: int) -> list[LangchainDocument]:
```

Comment:
Using `list[int]` is okay if the Python version is >=3.9. If not use the capital letter counterpart